query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java

```diff
@@ -59,7 +59,7 @@
 
    private static final int MAX_EXPANSION_COFACTORS = 16;
 
-   protected final AdvancedCache<?, ?> cache;
+   protected AdvancedCache<?, ?> cache;
```

Comment:
I'd rather keep this field final, and then do the ugly one liner in the constructor of RemoteQueryEngine